docs/recipes/install/poc-markdown/common/parse_doc_md.py

```diff
@@ -0,0 +1,314 @@
+        # Line continuation
+        elif match := self._PATTERNS["sms_continuation"].search(line):
+            self._process_continuation(match, output)
+        # Do loop
+        elif match := self._PATTERNS["sms_do_loop"].search(line):
+            self._process_do_loop(match, output)
+        # Regular SMS prompt
+        elif match := self._PATTERNS["sms_prompt"].search(line):
+            cmd = self._substitute_variables(match.group(1))
+            output.write(f"{cmd}\n")
+        # Plain comments
+        elif line.startswith("#"):
+            comment = self._substitute_variables(line)
+            output.write(f"{comment}\n")
+
+    def _substitute_variables(self, cmd: str) -> str:
+        """Substitute Jinja2-style variables with config values."""
+        # Handle {{ variable }} syntax
+        def replace_var(match):
+            var_name = match.group(1).strip()
+            return str(self.config.get(var_name, f"{{{{ {var_name} }}}}"))
+
+        # Replace Jinja2 variables
+        cmd = re.sub(r'\{\{\s*(\w+)\s*\}\}', replace_var, cmd)
+
+        return cmd
+
+    def _process_here_doc(self, match: re.Match, output: StringIO) -> None:
+        """Process HERE document."""
+        cmd = self._substitute_variables(match.group(1))
+        delimiter = match.group(2).strip()
+
+        output.write(f"{cmd}\n")
+
+        # Process remaining lines until delimiter
+        try:
+            while True:
+                line = next(self._line_iterator)
+                line = line.rstrip("\n\r").strip()
+
+                # Check if we hit code block end
+                if self._PATTERNS["code_block_end"].match(line):
+                    break
+
+                line = self._substitute_variables(line)
+                output.write(f"{line}\n")
+                if line.startswith(delimiter):
+                    break
+        except StopIteration:
+            pass
+
+    def _process_continuation(self, match: re.Match, output: StringIO) -> None:
+        """Process line continuation."""
+        cmd = self._substitute_variables(match.group(1))
+        output.write(f"{cmd} \\")
+
+        try:
+            while True:
+                line = next(self._line_iterator)
+                line = line.rstrip("\n\r").strip()
+
+                # Check for code block end
+                if self._PATTERNS["code_block_end"].match(line):
+                    output.write("\n")
+                    break
+
+                # Skip [sms]# prefix if present
+                if line.startswith("[sms]#"):
+                    line = line[6:].strip()
+
+                line = self._substitute_variables(line)
+
+                if line.endswith("\\"):
+                    output.write(f"\n\t{line}")
+                else:
+                    output.write(f"\n\t{line}\n")
+                    break
+        except StopIteration:
+            output.write("\n")
+
+    def _process_do_loop(self, match: re.Match, output: StringIO) -> None:
```

Comment:
I wonder if re-processing loops adds unneeded complexity, I had a lot of issues in the LaTeX version.  Perhaps the code should be formatted properly in the original markdown?